database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -212,11 +244,34 @@ public void write(DataOutput out) throws IOException {
    * @param stmt the {@link PreparedStatement} to write the {@link StructuredRecord} to
    */
   public void write(PreparedStatement stmt) throws SQLException {
+    List<String> updatedKeyList = new ArrayList<>();
+    if (relationTableKey != null) {
+      updatedKeyList = Arrays.asList(relationTableKey.split(","));
+    }
+    modifiableColumnTypes = new ArrayList<>(columnTypes);
     for (int i = 0; i < columnTypes.size(); i++) {
       ColumnType columnType = columnTypes.get(i);
       Schema.Field field = record.getSchema().getField(columnType.getName());
       writeToDB(stmt, field, i);
+      if (fillUpdateParams(updatedKeyList, columnType)) {
+        modifiableColumnTypes.add(columnType);
+      }
     }
+    if (operationName != null && relationTableKey != null) {
+      for (int i = columnTypes.size(); i < modifiableColumnTypes.size(); i++) {
+        ColumnType columnType = modifiableColumnTypes.get(i);
+        Schema.Field field = record.getSchema().getField(columnType.getName());
+        writeToDB(stmt, field, i);
+      }
```

Comment:
how Upsert will be perfomed here?

Reply:
in case of mysql plugin, this piece of code will be used for upsert as it will have question marks to be filled unlike other db plugins for upsert.

Reply:
this code was written to fill the question marks we get in case of update and upsert (only mysql based plugins).